pkg/util/conntrack/conntrack_test.go

```diff
@@ -17,301 +17,318 @@ limitations under the License.
 package conntrack
 
 import (
-	"fmt"
-	"strings"
+	"net"
+	"strconv"
 	"testing"
 
 	v1 "k8s.io/api/core/v1"
-	"k8s.io/utils/exec"
-	fakeexec "k8s.io/utils/exec/testing"
 	utilnet "k8s.io/utils/net"
+
+	"github.com/vishvananda/netlink"
 )
 
-func familyParamStr(isIPv6 bool) string {
-	if isIPv6 {
-		return " -f ipv6"
-	}
-	return ""
-}
+var ct conntrack
 
-func TestExecConntrackTool(t *testing.T) {
-	fcmd := fakeexec.FakeCmd{
-		CombinedOutputScript: []fakeexec.FakeAction{
-			func() ([]byte, []byte, error) { return []byte("1 flow entries have been deleted"), nil, nil },
-			func() ([]byte, []byte, error) { return []byte("1 flow entries have been deleted"), nil, nil },
-			func() ([]byte, []byte, error) {
-				return []byte(""), nil, fmt.Errorf("conntrack v1.4.2 (conntrack-tools): 0 flow entries have been deleted")
-			},
-		},
+func createUDPConnection(t *testing.T, srcIP, srcPort, dstIP, dstPort string) {
+	laddr, err := net.ResolveUDPAddr("udp", net.JoinHostPort(srcIP, srcPort))
+	if err != nil {
+		t.Fatalf("Unexpected error resolving address %s: %v", srcIP, err)
 	}
-	fexec := fakeexec.FakeExec{
-		CommandScript: []fakeexec.FakeCommandAction{
-			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(&fcmd, cmd, args...) },
-			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(&fcmd, cmd, args...) },
-			func(cmd string, args ...string) exec.Cmd { return fakeexec.InitFakeCmd(&fcmd, cmd, args...) },
-		},
-		LookPathFunc: func(cmd string) (string, error) { return cmd, nil },
+	raddr, err := net.ResolveUDPAddr("udp", net.JoinHostPort(dstIP, dstPort))
+	if err != nil {
+		t.Fatalf("Unexpected error resolving address %s: %v", srcIP, err)
 	}
-
-	testCases := [][]string{
-		{"-L", "-p", "udp"},
-		{"-D", "-p", "udp", "-d", "10.0.240.1"},
-		{"-D", "-p", "udp", "--orig-dst", "10.240.0.2", "--dst-nat", "10.0.10.2"},
+	conn, err := net.DialUDP("udp", laddr, raddr)
+	if err != nil {
+		t.Fatalf("Unexpected error creating the connection to %s: %v", raddr, err)
 	}
+	defer conn.Close()
+	_, err = conn.Write([]byte("knock knock knocking on heavens door"))
+	if err != nil {
+		t.Fatalf("Unexpected error sending datagram %s: %v", raddr, err)
+	}
+}
 
-	expectErr := []bool{false, false, true}
-
-	for i := range testCases {
-		err := Exec(&fexec, testCases[i]...)
+func TestClearEntriesForIP(t *testing.T) {
```

Comment:
This test require root privileges, otherwise it fails, is it possible to give it those privileges or should I mock it?
It will be great being able to test this with real flows though

Reply:
We do run as root in CI currently but it's kinda not great for local developers ...

Reply:
it needs CAP_NET_ADMIN, maybe the pod that is running does not allow it despite being root?